### PR TITLE
Scala.js support in macroUtils, tagged and opaque

### PR DESCRIPTION
**Pull request by adpi2**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -16,7 +16,7 @@ lazy val baseSettings = Seq(
   organizationName := "Iterators",
   organizationHomepage := Some(url("https://iterato.rs")),
   homepage := Some(url("https://github.com/theiterators/kebs")),
-  scalacOptions := Seq("-deprecation", "-unchecked", "-feature", "-encoding", "utf8")
+  scalacOptions ++= Seq("-deprecation", "-unchecked", "-feature", "-encoding", "utf8")
```
Comment (adpi2) [inline build.sbt:19]: The `sbt-scalajs` plugin also adds some `scalacOptions`.

File: build.sbt [hunk 2]
```diff
@@ -104,20 +104,19 @@ def paradisePlugin(scalaVersion: String): Seq[ModuleID] =
   else
     Seq.empty
 
-val scalaTest       = "org.scalatest" %% "scalatest" % "3.2.14"
-val scalaCheck      = "org.scalacheck" %% "scalacheck" % "1.17.0"
+val scalaTest       = Def.setting("org.scalatest" %%% "scalatest" % "3.2.14")
```
Comment (adpi2) [inline build.sbt:107]: The `%%%` needs to acces the Scala.js settings, so we need to enclose that in a setting too.

**Review by adpi2 — COMMENTED** [review]
This PR adds Scala.js support in `macroUitls`, `tagged` and `opaque`.

Slick is not available on Scala.js so there is no `SlickSupport` in `tagged` on Scala.js. As a consequece there is no tests on `tagged` for Scala.js.

I was not able to cross compile other projects with Scala.js because of the dependency to `instances` which uses Java time and `Currency` from the JDK. Those are not implemented in the JDK implementation of Scala.js. For Java time we could use [cquiroz/scala-java-time](https://index.scala-lang.org/cquiroz/scala-java-time).

**Comment by luksow** [comment]
@pk044 can you review & merge please?

We should also ASAP look into https://github.com/theiterators/kebs/issues/212 as it's gonna be more problematic going forward (and onto kebs 2).

**Review by pk044 — APPROVED** [review]

**Comment by pk044** [comment]
@adpi2 thank you for the PR! :pray:

**Pull request merged**